File: Model-View-ViewModel-Java/app/src/main/java/com/mvvm/java/core/data/network/ApiServices.java
package com.mvvm.java.core.data.network;

import com.mvvm.java.core.data.network.models.response.ModelResponse;

import java.util.List;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiServices {
    @GET(ApiEndpoint.GET_USERS)
    Single<List<ModelResponse>> getUsers();

    @GET(ApiEndpoint.GET_USERS)
    Single<List<ModelResponse>> getUsers(@Query("page") int pageIndex);

    @GET(ApiEndpoint.GET_USER_DETAIL)
    Single<ModelResponse> getUserDetail(@Path("userId") String userId);
}

File: Android-MVVM-Kotlin/app/src/main/java/com/base/mvvm/core/data/network/ApiServices.kt
package com.base.mvvm.core.data.network

import com.base.mvvm.core.data.network.models.ApiUser
import retrofit2.http.GET

/**
 * Author: William Giang Nguyen | 8/7/2022
 * */
interface ApiServices {
    @GET(ApiEndpoint.GET_USER_LIST)
    fun getUsers(): List<ApiUser>

    @GET(ApiEndpoint.GET_MORE_USERS)
    suspend fun getMoreUsers(): List<ApiUser>

    @GET(ApiEndpoint.GET_USERS_WITH_ERROR)
    suspend fun getUsersWithError(): List<ApiUser>
}
